src/elements/content-sharing/__tests__/ContentSharing.test.js

```diff
@@ -0,0 +1,339 @@
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { mount } from 'enzyme';
+import { FormattedMessage } from 'react-intl';
+import API from '../../../api';
+import ErrorMask from '../../../components/error-mask/ErrorMask';
+import ContentSharing from '../ContentSharing';
+import UnifiedShareModal from '../../../features/unified-share-modal/UnifiedShareModal';
+import { DEFAULT_HOSTNAME_API, TYPE_FILE, TYPE_FOLDER } from '../../../constants';
+import { convertItemResponse, convertUserResponse } from '../../../features/unified-share-modal/utils/convertData';
+import {
+    MOCK_ITEM,
+    MOCK_ITEM_API_RESPONSE,
+    MOCK_CONVERTED_ITEM_DATA,
+    MOCK_CONVERTED_USER_DATA,
+    MOCK_SHARED_LINK,
+    MOCK_SHARED_LINK_DATA_AFTER_NORMALIZATION,
+    MOCK_USER_API_RESPONSE,
+} from '../../../features/unified-share-modal/utils/__mocks__/USMMocks';
+
+jest.mock('../../../api');
+jest.mock('../../../features/unified-share-modal/utils/convertData');
+
+describe('elements/unified-share-modal-element/ContentSharing', () => {
+    const getWrapper = props =>
+        mount(<ContentSharing apiHost={DEFAULT_HOSTNAME_API} itemID="" language="" token="" {...props} />);
+
+    beforeEach(() => {
+        convertItemResponse.mockReturnValue(MOCK_CONVERTED_ITEM_DATA);
+        convertUserResponse.mockReturnValue(MOCK_CONVERTED_USER_DATA);
+    });
+
+    afterEach(() => {
+        jest.resetModules();
+        jest.restoreAllMocks();
+    });
+
+    describe('with successful API calls', () => {
+        test('should call getFileAPI().getFile() if itemType is "file"', async () => {
+            const getFile = jest.fn().mockImplementation((id, successFn) => {
+                return Promise.resolve(MOCK_ITEM_API_RESPONSE).then(response => {
+                    successFn(response);
+                });
+            });
+            API.mockImplementation(() => ({
+                getFileAPI: jest.fn().mockReturnValue({ getFile }),
+                getUsersAPI: jest.fn().mockReturnValue({ getUser: jest.fn() }),
+            }));
+
+            let wrapper;
+            await act(async () => {
+                wrapper = getWrapper({ itemType: TYPE_FILE });
+            });
+
+            wrapper.update();
+            const usm = wrapper.find(UnifiedShareModal);
+
+            expect(getFile).toHaveBeenCalled();
+            expect(convertItemResponse).toHaveBeenCalledWith(MOCK_ITEM_API_RESPONSE);
+            expect(usm.length).toBe(1);
```

Comment:
Can be replaced with `wrapper.exists(UnifiedShareModal)`, if desired. Since we have other assertions that rely on it existing, though, it seems like this isn't necessary.